keps/sig-auth/3766-referential-authorization/README.md

```diff
@@ -259,66 +239,76 @@ the API:
-is removed as data points.
-
 ### Example Usage
 
-#### Gateway API Gateway Referencing Secret
+#### Gateway API Cross-Namespace Secret Reference
 
-In this example (from the Gateway API docs), we have a Gateway in the
-`gateway-api-example-ns1` namespace, referencing a Secret in the
-`gateway-api-example-ns2` namespace. The following ReferenceGrant allows this:
+Authors of APIs can bundle ClusterReferencePatterns with their API definitions
+for any references they expect to be common. For example, Gateway API could
+bundle the following ClusterReferencePattern to describe how Gateways reference
+TLS Secrets. Note that this also allows API authors to define a `SameNamespace`
+`baselineGrant` that would automatically authorize all references following this
+pattern without crossing namespace boundaries.
 
 ```yaml
-apiVersion: gateway.networking.k8s.io/v1beta1
-kind: Gateway
+kind: ClusterReferencePattern
+apiVersion: reference.authorization.k8s.io/v1alpha1
 metadata:
-  name: cross-namespace-tls-gateway
-  namespace: gateway-api-example-ns1
-spec:
-  gatewayClassName: acme-lb
-  listeners:
-  - name: https
-    protocol: HTTPS
-    port: 443
-    hostname: "*.example.com"
-    tls:
-      # There's a Kind/Resource mismatch here, which sucks, but it is not
-      # easily fixable, since Gateway is already a beta, close to GA
-      # object.
-      certificateRefs:
-      - kind: Secret
-        group: ""
-        name: wildcard-example-com-cert
-        namespace: gateway-api-example-ns2
----
-apiVersion: authorization.k8s.io/v1alpha1
+  name: gateway-tls
+group: gateway.networking.k8s.io
+resource: gateways
+version: v1
+path: ".spec.listeners[*].tls.certificateRefs[*]"
+baselineGrant: SameNamespace
+```
+
+Implementations of APIs can bundle ClusterReferenceConsumer resources as part of
+their deployment. This resource links an RBAC Subject (likely the ServiceAccount
+used by the controller) to a ClusterReferencePattern. Once linked, controllers
+will gain access to anything that has been granted following that pattern.
+
+```yaml
+kind: ClusterReferenceConsumer
+apiVersion: reference.authorization.k8s.io/v1alpha1
+metadata:
+  name: contour-gateway
+subject:
+  kind: ServiceAccount
+  name: contour
+  namespace: contour-system
+patternNames:
+- gateway-tls
+```
+
+Finally, users can use ReferenceGrant to authorize specific references that
+follow a ClusterReferencePattern. For example, the following ReferenceGrant
+authorizes references from Gateways in the `prod` namespace to the `acme-tls`
+Secret in the `prod-tls` namespace:
+
+```yaml
 kind: ReferenceGrant
+apiVersion: reference.authorization.k8s.io/v1alpha1
 metadata:
-  name: allow-ns1-gateways-to-ref-secrets
-  namespace: gateway-api-example-ns2
-spec:
-  from:
-  - group: gateway.networking.k8s.io
-    resource: gateways
-    namespace: gateway-api-example-ns1
-  to:
-  - group: ""
-    resource: secrets
+  name: prod-gateways
+  namespace: prod-tls
+patternName: gateway-tls
```

Comment:
May be clearer:
```suggestion
patternName: gateway-tls # assume that there is a suitable ClusterReferencePattern
                         # (see earlier example)
```

Reply:
Yeah this was a common point of confusion, pushed up an update that removes the link to a pattern altogether here.